libtenzir/builtins/formats/lines.cpp

```diff
@@ -106,44 +108,133 @@ class lines_parser final : public plugin_parser {
   parser_args args_;
 };
 
-// To get better diagnostics in `write`, i.e.:
-//    "'xsv' does not support heterogeneous outputs..."
-// -> "'lines' does not..."
-// Otherwise, we could just use the `ssv_plugin` directly
-class lines_printer final : public plugin_printer {
-public:
-  lines_printer() = default;
-
-  lines_printer(std::unique_ptr<plugin_printer> inner)
-    : inner_(std::move(inner)) {
+struct lines_printer_impl {
+  template <typename It>
+  auto print_values(It& out, const view<record>& x) const noexcept -> bool {
+    auto first = true;
+    for (const auto& [_, v] : x) {
+      if (!first) {
+        ++out = ' ';
+      } else {
+        first = false;
+      }
+      caf::visit(visitor{out}, v);
+    }
+    return true;
   }
 
+  template <class Iterator>
+  struct visitor {
+    visitor(Iterator& out) : out{out} {
+    }
+
+    auto operator()(caf::none_t) noexcept -> bool {
+      sequence_empty = false;
+      *out++ = '-'; // TODO: Should null values be mapped to empty string?
```

Comment:
Another option would be not to render null values at all, i.e., to just skip over them.

Reply:
Yes, that is what I meant.

Reply:
Don't think so, should be safe to do.